### PR TITLE
New helper func to set KEDA autoscaling annotations for kafka resources

### PR DESCRIPTION
**Pull request by aavarghese**
Signed-off-by: aavarghese <avarghese@us.ibm.com>

As per title and review feedback from @pierDipi.

### CODE REVIEW

File: control-plane/pkg/keda/keda.go [hunk 1]
```diff
@@ -204,3 +204,25 @@ func addAuthSecretTargetRef(parameter string, secretKeyRef v1beta1.SecretValueFr
...
+
+func SetAutoscalingAnnotations(objannotations map[string]string) map[string]string {
+	if objannotations != nil {
+		cgannotations := map[string]string{}
+		setAnnotation(objannotations, keda.AutoscalingClassAnnotation, cgannotations)
+		setAnnotation(objannotations, keda.AutoscalingMinScaleAnnotation, cgannotations)
+		setAnnotation(objannotations, keda.AutoscalingMaxScaleAnnotation, cgannotations)
+		setAnnotation(objannotations, keda.KedaAutoscalingPollingIntervalAnnotation, cgannotations)
+		setAnnotation(objannotations, keda.KedaAutoscalingCooldownPeriodAnnotation, cgannotations)
+		setAnnotation(objannotations, keda.KedaAutoscalingKafkaLagThreshold, cgannotations)
```
Comment (zroubalik) [inline control-plane/pkg/keda/keda.go:216]: Just FYI, a new setting: activationLagThreshold has been introduced in KEDA 2.8.1, adding to eventing-autoscaler-keda here: https://github.com/knative-sandbox/eventing-autoscaler-keda/pull/329

**Review by zroubalik — APPROVED** [review]
I got lost in these PR, so putting the same remark here.


/lgtm

File: control-plane/pkg/keda/keda.go (hunk 1 above)
Comment (aavarghese) [reply control-plane/pkg/keda/keda.go:216]: @zroubalik sorry for sending you on a maze hunt with these 2 PRs. Plan was that this current PR (with helper code) would have merged prior to the [KafkaSource one](https://github.com/knative-sandbox/eventing-kafka-broker/pull/2608). But looks like the latter will beat this one in the merge pool.

So marking this PR as not relevant anymore and closing it.
I've noted your comment with the KafkaSource PR and will be implementing that soon. Thank you!

**Comment by aavarghese** [comment]
/hold
/close

Changes in here are included in https://github.com/knative-sandbox/eventing-kafka-broker/pull/2608 so not needed.

**Pull request closed without merging**